src/app/@ansyn/plugins/openlayers/maps/open-layers-map/openlayers-map/openlayers-map.ts

```diff
@@ -32,24 +43,54 @@ export enum StaticGroupsKeys {
 // @dynamic
 @ImageryMap({
 	mapType: OpenlayersMapName,
-	deps: [OpenLayersProjectionService, CoreConfig],
+	deps: [OpenLayersProjectionService, CoreConfig, Actions, Store],
 	defaultMapSource: 'BING'
 })
 export class OpenLayersMap extends BaseImageryMap<OLMap> {
 	static groupsKeys = StaticGroupsKeys;
 	static groupLayers = new Map<StaticGroupsKeys, Group>(Object.values(StaticGroupsKeys).map((key) => [key, new Group()]) as any);
 	private showGroups = new Map<StaticGroupsKeys, boolean>();
 	private _mapObject: OLMap;
+	private _backgroundMapObject: OLMap;
+	private _backgroundMapParams: object;
+
 	private _moveEndListener: () => void;
 	private olGeoJSON: OLGeoJSON = new OLGeoJSON();
 	private _mapLayers = [];
 	public isValidPosition;
 	public shadowElement = null;
+	private savedParams: any;
+	private mapId: string;
+
+	public isLoadingLayers$: Observable<boolean> = this.store$.select(selectIsLoadingTiles).pipe(
+		map((f) => f(this.mapId))
+	);
+
+	setMainLayerToForegroundMapAfterTilesAreLoaded() {
+		this.actions$.pipe(
+			ofType<SetProgressBarAction>(MapActionTypes.VIEW.SET_PROGRESS_BAR),
+			filter(({ payload }) => {
+				// console.log('progress', payload.progress);
+				return payload.progress === 100;
+			}),
+			debounceTime(500), // Adding debounce, to compensate for strange multiple loads when reading tiles from the browser cache (e.g. after browser refresh)
+			tap(({ payload }) => {
+				const { layer, view, position, extent, rotation } = this.savedParams;
+				this.setMainLayerToForegroundMap(layer);
+				this._mapObject.setView(view);
+				this._setMapPositionOrExtent(this.mapObject, position, extent, rotation).pipe(take(1)).subscribe();
+			}),
+			take(1)
```

Comment:
add take until of configured time -> 3 sec for start

Reply:
done